src/app/clusters/smoke-co-alarm-server/smoke-co-alarm-server.cpp

```diff
@@ -55,28 +55,28 @@ void SmokeCoAlarmServer::SetExpressedStateByPriority(EndpointId endpointId,
         switch (priority)
         {
         case ExpressedStateEnum::kSmokeAlarm:
-            VerifyOrReturn(GetSmokeState(endpointId, alarmState));
+            GetSmokeState(endpointId, alarmState);
```

Comment:
So this assumes that the caller included kSmokeAlarm in priorityOrder even though their alarm does not implement SMOKE?

That seems like a really weird thing to do, but documenting here that we are ignoring the return value to handle that case would be good.

But also, this is assuming that on failure the getter does not modify alarmState, and nothing promises that, right?    Seems like what we should be doing is:

1. alarmState/endOfServiceState/active should be declared inside the loop, so they are reset to the default values on every iteration.
2. Failures from these getters should immediately continue the loop, instead of falling through to examining possibly-garbage values.
3. Why failures continue immediately should be documented.